merge_checks.py

```diff
@@ -28,16 +32,27 @@ def get_base_revision(base_ref: str) -> str:
     return _run_process(f"git rev-parse origin/{base_ref}")
 
 
-def get_subject_markers(head_hash: str, base_hash: str) -> Tuple[str, ...]:
-    log = _run_process(f"git log --pretty='format:%s' {base_hash}..{head_hash}").splitlines()
-    return tuple(line.split(maxsplit=1)[0] for line in log)
+def get_subject(head_hash: str, base_hash: str) -> Sequence[str]:
+    return tuple(_run_process(f"git log --pretty='format:%s' {base_hash}..{head_hash}").splitlines())
+
+
+def get_subject_markers(subjects: Sequence[str]) -> Sequence[str]:
+    return tuple(line.split(maxsplit=1)[0] for line in subjects)
 
 
 def has_merge_commits(head_hash: str, base_hash: str) -> bool:
     parents = _run_process(f"git log --pretty=%p {base_hash}..{head_hash}").splitlines()
     return any(len(line.split(maxsplit=1)) > 1 for line in parents)
 
 
+def has_wrong_commit_message(subject_markers: Sequence[str]) -> Sequence[str]:
+    return tuple(
+        marker
+        for marker in subject_markers
+        if re.match(rf"^({'|'.join(ALLOWED_COMMIT_MESSAGE_TYPES)})((\([a-z]*\): .*)|(: .*))", marker) is None
```

Comment:
Note that `[a-z]*` allows for an empty thing, potentially allowing messages like `ci(): foo` which feels weird... I would suggest to at least set it to `[a-z]+`.

Also it won't let things like `perf(global-redirects)` because `[a-z]` does not include the dash.